src/arch/arm/tlb.cc

```diff
@@ -240,98 +228,75 @@ TLB::checkPromotion(TlbEntry *entry, BaseMMU::Mode mode)
 
 // insert a new TLB entry
 void
-TLB::insert(TlbEntry &entry)
+TLB::insert(const Lookup &lookup_data, TlbEntry &entry)
```

Comment:
Why do we add another parameter instead of using entry->keyType?

Reply:
Interesting I will consider this option...

Reply:
So I looked into it and I am not fully convinced.. The issue is that KeyAccess does not convey the tag information only, but it contains access specific variables, like the aforementioned functional mentioned above. This means the TlbEntry might end up with garbage data within it

Reply:
This, for me, is a hint that the KeyType may contain more stuff than it should. Nonetheless, I will disregard this request for now

Reply:
Thanks @odanrc !